substrate/frame/revive/rpc/src/subxt_client.rs

```diff
@@ -0,0 +1,71 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//! The generated subxt client.
+//! Generated with
+//! subxt metadata  --url ws://localhost:9944 -o rpc/kitchensink.scale
+use subxt::config::{signed_extensions, Config, PolkadotConfig};
+
+#[subxt::subxt(
+	runtime_metadata_path = "kitchensink.scale",
```

Comment:
This will not work against `Westend`, right? So this seems like a non starter to be tied to a specific runtime.

Reply:
testing this now with zombienet,
I think it will complain that metadata are not compatible,
for now the workaround would be to put each supported chain metadata behind a feature flag

Reply:
> I think it will complain that metadata are not compatible,

For a reason. The pallet ids will not match and you will be dispatching garbage. 

> for now the workaround would be to put each supported chain metadata behind a feature flag

Yeah this is fine for now.

Reply:
I thought subxt can also fetch the metadata so we don't have to vendor it?

Reply:
Yeah but it will download at build time since it is used for codegen. Might be an alternative for westend which has a public URL.

Reply:
Asked James, this is fine, we can generate the md from asset-hub (which I did) and it will work with chains so long that chain's types are the same.
Tested the RPC against kitchensink and AH (via  zombienet ) it does work.

>Code generated for one chain should work just fine with another chain so long as the types and such are all the same (and if they aren't, you'll get an "incompatible codegen" error when you try to do the thing that isn't compatible)